src/registry.rs

```diff
@@ -183,17 +185,17 @@ impl<M> Registry<M> {
     /// # use prometheus_client::metrics::counter::{Atomic as _, Counter};
     /// # use prometheus_client::registry::{Registry, Unit};
     /// #
-    /// let mut registry: Registry<Counter> = Registry::default();
+    /// let mut registry = Registry::default();
     ///
-    /// let subsystem_a_counter_1 = Counter::default();
-    /// let subsystem_a_counter_2 = Counter::default();
+    /// let subsystem_a_counter_1: Counter = Counter::default();
+    /// let subsystem_a_counter_2: Counter = Counter::default();
```

Comment:
These type ascriptions are unfortunate. Can we get rid of them somehow?

Reply:
As far as I can tell, this is due to the default value of the `Counter` `A` trait parameter.

One thought I had was reducing the number of indirections through a `Counter::new` method, though unfortunately, the compiler can not infer `A` in such case either.